framerail/src/lib/server/load/login.ts

```diff
@@ -0,0 +1,44 @@
+import defaults from "$lib/defaults"
+import { translate } from "$lib/server/deepwell/translate"
+import type { TranslateKeys } from "$lib/types"
+import { parse } from "accept-language-parser"
+
+export async function loadLoginPage(request, cookies) {
+  // Set up parameters
+  const url = new URL(request.url)
+  const domain = url.hostname
+  const sessionToken = cookies.get("wikijump_token")
+  const language = request.headers.get("Accept-Language")
+  let locales = parse(language)
+    .sort((a, b) => a.quality - b.quality)
+    .map((lang) => (lang.region ? `${lang.code}-${lang.region}` : lang.code))
```

Comment:
I appreciate the sorting, though this version of locale reconstruction still discards the `script` value.

Basically the implementation is:
```typescript
function rebuildLocale(lang: object): string {
    const parts = [lang.code]
    if (lang.script) parts.push(lang.script)
    if (lang.region) parts.push(lang.region)
    return parts.join('-')
}

[previous code]
.map(rebuildLocale)
```

Reply:
I looked at the source code of `accept-language-parser`, seems that they didn't mention in the readme but they do parse script, and they already apply sorting in descending quality value. I will still keep the sorting in the code though.